### PR TITLE
Moa and Karan - Week 20 - Project Authentication

### PR DESCRIPTION
**Pull request by karanmann**
# Link to my deployed project:

- FrontEnd : wk20-api-auth.netlify.app
- BackEnd : https://login-logout-authentication.herokuapp.com/

# My reflections on how this week's project turned out:

This project was fun and really challenging. The scope of the project was much bigger than the previous weeks which was a bit of a surprise. It was also great to work as a team as this was our last project in the course.

# We aimed for the blue level this time around.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,28 @@
...
+- Heroku
+- React
+- React Router
+- Redux - Toolkit + Hooks
+- JS
 
-Every project should be deployed somewhere. Be sure to include the link to the deployed project so that the viewer can click around and see what it's all about.
+# View it live 🔴
+
+- FrontEnd : wk20-api-auth.netlify.app
```
Comment (sandra-shumanteva-kaevska) [inline README.md:27]: Maybe it is better to write full link which is clickable :)

File: backend/server.js [hunk 2]
```diff
@@ -2,28 +2,95 @@ import express from 'express'
...
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
+const User = mongoose.model('User', {
+  name: {
+    type: String, 
+    unique: true,
+    required: true,
+  },
+  email: {
```
Comment (sandra-shumanteva-kaevska) [inline backend/server.js:18]: Well done with adding the email in sign-in  form 💯

File: backend/server.js [hunk 3]
```diff
@@ -2,28 +2,95 @@ import express from 'express'
...
+    await user.save()
+    res.status(201).json({ userId: user._id, accessToken: user.accessToken })
+  } catch (err) {
+    res.status(400).json({ message: 'Could not create user', errors: err.errors })
+  }
+})
+
+app.get('/secrets', authenticateUser)
+app.get('/secrets', async (req, res) => {
+  const secretMessage = `This is a super secret message for ${req.user.name}`
```
Comment (sandra-shumanteva-kaevska) [inline backend/server.js:81]: When you login and reveal the secret message it is showing like this: {"secretMessage":"This is a super secret message for test"}
To fix this you need to wrap your template string in {} like this: const secretMessage = {`This is a super secret message for ${req.user.name}`}

**Review by sandra-shumanteva-kaevska — APPROVED** [review]
Great Job Moa and Karan you achieved all blue level requirements. The code was clean and easy to read. Keep up with the good work :D

**Review by KarolCislowski — CHANGES_REQUESTED** [review]
This app has a very big potential, but you need to do 2 things:

- store your user in the local storage to stop logging out on page refresh,
- add `_redirects` file to the `public` directory inside of your frontend and save the following code  in it: `/* /index.html 200`   It will stop netlify from showing a default error page.

And the last thing: Please, make yourself a favour and style your app nicely. It should take only a few minutes but it can open some extra doors 😄

**Comment by karanmann** [comment]
Hej Karol,

Thank you for the feedback. We have made the requested changes as you mentioned in your bullet points.
We have decided to keep the styling as it is for now and go back to it later. And the same goes for the signup to login.
We do understand it would be a good UX but when we created it we thought this was the best practice.  

/ Moa and Karan

**Review by KarolCislowski — APPROVED** [review]
Now it works nice 😺  Great job 🦄

**Pull request closed without merging**